### ISSUE TITLE
Support for OpenID SSO

### ISSUE THREAD

**Issue by skid9000** [syuilo/Misskey#9132]
## Summary

Hello !
Is it planned to allow authentication via an SSO provider (via OpenID standard) ?

**Comment by FunctionalHacker** [comment]
Seconded! Single Sign-On with OIDC/OAuth would be very useful since I host my own Authentik instance

**Comment by sussycatgirl** [comment]
Just chiming in to say that I'm interested in this too!

**Comment by rxy** [comment]
related or dup with #8262

**Comment by gnh1201** [comment]
@rxy related but no dup. Sometimes, these two perform different purposes within the same protocol.

**Comment by sftblw** [comment]
I'm not sure but by my knowledge, it's similar to "Login with Github, Google, etc..." but OIDC is a standardized version of it.

| .                     | Misskey          | Target Service                                                                                     |
|-----------------------|------------------|----------------------------------------------------------------------------------------------------|
| OAuth Provider (8262) | OAuth 2.0 Server | OAuth 2.0 Client                                                                                   |
| Login with Github     | OAuth 2.0 Client | OAuth 2.0 Server                                                                                   |
| OpenID Connect (OIDC) | OAuth 2.0 Client<br/>with support of OIDC spec<br/>utilizing target service<br/>as a identity provider | OAuth 2.0 Server<br/>with predefined OIDC informations<br/>of user identities<br/><br/>examples:<ul><li>Azure Active Directory</li><li>Google Identity Platform</li><li>Keycloak</li><li>Authentik</li></ul> |

**Comment by tetradice** [comment]
Great request!

It was difficult to find when I searched in Japanese, so I will list the keywords for searching.

Googleログイン
Facebookログイン
Twitterログイン
Githubログイン
SNSログイン

**Comment by bil-ash** [comment]
Yeah, I would also like to provide users with the option to sign-up via sso-for the moment only google,facebook and twitter

**Comment by Mrs-Feathers** [comment]
i would also enjoy this! more support!

**Comment by zxcv-12345** [comment]
It seems that there is no official desire to support single sign-on at this time.

**Comment by bil-ash** [comment]
@syuilo Any plans for supporting misskey as a oauth 2.0 client. Servers like Nodebb and Mbin already support registering an account using oauth 2.0 client (google,github,etc). This actually reduces user on-boarding friction, more so for non-technical users, so would be nice to have it supported in misskey also.

**Comment by bil-ash** [comment]
@samunohito Any updates?

**Comment by f4rr0l1v3r** [comment]
Hi! Is this in development or open without progres? 

Thank you!
